tests/src/3d/sandbox/qgis_3d_sandbox.cpp

```diff
@@ -106,6 +115,69 @@ void initCanvas3D( Qgs3DMapCanvas *canvas )
   qDebug() << "pending jobs:" << canvas->scene()->totalPendingJobsCount();
 }
 
+QPointer<QDialog> createConfigDialog( Qgs3DMapCanvas *canvas )
+{
+  const QPointer configDialog = new QDialog;
+  configDialog->setWindowTitle( QStringLiteral( "3D Configuration" ) );
+  configDialog->setObjectName( QStringLiteral( "3DConfigurationDialog" ) );
+  configDialog->setMinimumSize( 600, 460 );
+  QgsGui::enableAutoGeometryRestore( configDialog );
+
+  Qgs3DMapSettings *map = canvas->mapSettings();
+  Qgs3DMapConfigWidget *w = new Qgs3DMapConfigWidget( map, new QgsMapCanvas, canvas, configDialog );
```

Comment:
Rather than creating a `QgsMapCanvas` here, I would suggest to pass nullptr here, and make the 2D map canvas optional in Qgs3DMapConfigWidget (it seems to be used only in one place anyway)